apollo-ios-codegen/Sources/GraphQLCompiler/CompilationResult.swift

```diff
@@ -139,12 +142,19 @@ public final class CompilationResult: JavaScriptObjectDecodable {
       self.variables = variables
       self.rootType = rootType
       self.selectionSet = selectionSet
-      self.directives = directives
       self.referencedFragments = referencedFragments
       self.source = source
       self.filePath = filePath
       self.isLocalCacheMutation = directives?
         .contains { $0.name == Constants.DirectiveNames.LocalCacheMutation } ?? false
+        
+      let referencedImports: [Directive] = self.referencedFragments
+        .compactMap { $0.directives?.filter { $0.name == Constants.DirectiveNames.Import } }
+        .flatMap { $0 }
+      self.directives = Array(Set(directives ?? [] + referencedImports))
+
+      self.moduleImports = self.directives?
```

Comment:
I'm also wondering if we should alphabetize these so that changing the order of Fragment Spreads doesn't change the order of import statements. That could be confusing when you change something in one Fragment definition and then a bunch of operations have their `import` statements change.

I see the use of `Set` when setting up the `directives` here. But if we are going to not add the referenced fragment imports to the `directives`, we should probably just make `moduleImports` an `OrderedSet` to ensure we don't get duplicate import statements.